Translation/Loader/XliffLoader.php

```diff
@@ -59,6 +59,23 @@ public function load($resource, $locale, $domain = 'messages')
                     ->setDesc((string) $trans->source)
                     ->setLocaleString((string) $trans->target)
             ;
+
+            $m->setApproved($trans['approved']=='yes');
+
+            if (isset($trans->target['state'])) {
+                $m->setState($trans->target['state']);
+            }
+
+            if (isset($trans->note)) {
+                if (count($trans->note) > 1) {
+                    foreach ($trans->note as $note) {
+                        $m->addNote((string) $note, isset($note['from']) ? ((string) $note['from']) : null);
+                    }
+                } else {
+                    $m->addNote((string) $trans->note, isset($trans->note['from']) ? ((string) $trans->note['from']) : null);
+                }
```

Comment:
so if they are testing is_array... but then else the block of code both does a cast to string, **and** an isset on the from. Which makes this part a bit confusing. If this is legitimately the best way to do what it is doing some inline comments would surely help future devs understand it better